### PR TITLE
Force Zip file extraction to use UTF-8 for netfx

### PR DESCRIPTION
**Pull request by rain-on**
If a Zip file does not explicitly indicate to use UTF-8 for decoding (via the EFS flag) - the file/directory entry will be extracted using the system-default encoding (which may not support multi-byte chars).

When using netcore - if no encoding is specified, it ALWAYS uses UTF-8 - as such, this issue only affects netfx users of calamari.

### CODE REVIEW

**Comment by APErebus** [comment]
Are we looking to backport this? If so, probably easier to target the release branch first then forward merge to `main`

Also, what happens if the _zip_ filename has accent chars in it? Might be a good second test

**Comment by rain-on** [comment]
I would like to backport.
I had assumed we'd update calamari as a standard commit - then apply the new calamari version to the old octopus - then forward merge?

File: source/Calamari.Tests/Fixtures/Integration/Packages/Samples/utf8.Filenámes-1.0.0.zip
Comment (kevjt) [inline source/Calamari.Tests/Fixtures/Integration/Packages/Samples/utf8.Filenámes-1.0.0.zip:1]: What's this file used for? I see the test only references the other one. Are you missing a test case maybe?

File: source/Calamari.Tests/Fixtures/Integration/Packages/PackageExtractorFixture.cs [hunk 1]
```diff
@@ -47,6 +47,27 @@ public void ExtractPumpsFilesToFilesystem(Type extractorType, string extension,
...
+        
+        
+        [Test]
+        public void CanExtractZipFileContainingSpecialCharacters()
+        {
+            var fileName = GetFixtureResource("Samples", string.Format("utf8.Filenames-1.0.0.zip"));
+
+            Type extractorType = typeof(ZipPackageExtractor);
+            
+            var extractor = (IPackageExtractor) Activator.CreateInstance(extractorType, ConsoleLog.Instance);
```
Comment (kevjt) [inline source/Calamari.Tests/Fixtures/Integration/Packages/PackageExtractorFixture.cs:59]: Is there any reason why the extractor can't be instantiated directly with `new`? That way you don't need to have the additional public constructor just for this test.

**Review by kevjt — COMMENTED** [review]
Looks mostly good. Some questions

**Comment by flin-8** [comment]
Changes look good to me, Trent will rebase off the Calamari branch for 2024.3

File: source/Calamari.Tests/Fixtures/Integration/Packages/Samples/utf8.Filenámes-1.0.0.zip
Comment (rain-on) [reply source/Calamari.Tests/Fixtures/Integration/Packages/Samples/utf8.Filenámes-1.0.0.zip:1]: missed a file - the same test now can unzip either file - just to confirm that utf-8 chars in the zip filename are handled.

File: source/Calamari.Tests/Fixtures/Integration/Packages/PackageExtractorFixture.cs (hunk 1 above)
Comment (rain-on) [reply source/Calamari.Tests/Fixtures/Integration/Packages/PackageExtractorFixture.cs:61]: way better solution, 🙏

**Review by kevjt — APPROVED** [review]
LGTM

**Comment by rain-on** [comment]
Closed, and reimplemented on an old branch, which will be merged forward.

**Pull request closed without merging**